rust/src/transformations/quantile_score_candidates/mod.rs

```diff
@@ -178,141 +220,47 @@ pub(crate) fn score_candidates_map<T: Integer + ExactIntCast<IntDistance>>(
 /// * `alpha_num` - numerator of alpha fraction
 /// * `alpha_den` - denominator of alpha fraction. alpha fraction is {0: min, 0.5: median, 1: max, ...}
 /// * `size_limit` - maximum size of `x`. If `x` is larger than `size_limit`, scores are truncated
-pub(crate) fn compute_score<TIA: PartialOrd>(
-    mut x: Vec<TIA>,
-    candidates: &Vec<TIA>,
-    alpha_num: usize,
-    alpha_den: usize,
-    size_limit: usize,
-) -> Vec<usize> {
-    // x must be sorted because counts are done via binary search
-    x.sort_by(|a, b| a.partial_cmp(&b).unwrap_or(Ordering::Equal));
-
-    // compute #(`x` < c) and #(`x` = c) for each c in candidates
-    let mut num_lt = vec![0; candidates.len()];
-    let mut num_eq = vec![0; candidates.len()];
-    count_lt_eq_recursive(
-        num_lt.as_mut_slice(),
-        num_eq.as_mut_slice(),
-        candidates.as_slice(),
-        x.as_slice(),
-        0,
-    );
-
-    // now that we have num_lt and num_eq for each candidate, score all candidates
-    num_lt
-        .into_iter()
-        .zip(num_eq.into_iter())
-        // score function cannot overflow.
-        //     lt <= size_limit, so 0 <= alpha_denom * lt <= usize::MAX
-        //     n - eq <= size_limit, so 0 <= size_limit - eq
-        .map(|(lt, eq)| {
-            // |α_den * #(x < c) - α_num * (n - #(x = c))|
-            (alpha_den * lt.min(size_limit)).abs_diff(alpha_num * (x.len() - eq).min(size_limit))
-        })
-        .collect()
-}
-
-/// Compute number of elements less than and equal to each edge
-/// Formula is (#(`x` < `e`), #(`x` == `e`)) for each e in `edges`.
-///
-/// # Arguments
-/// * `count_lt` - location to write the count of elements less than each edge
-/// * `count_eq` - location to write the count of elements equal to each edge
-/// * `edges` - edges to collect counts for. Must be sorted
-/// * `x` - dataset to count against
-/// * `x_start_idx` - value to add to the count. Useful for recursion on subslices
-fn count_lt_eq_recursive<TI: PartialOrd>(
-    count_lt: &mut [usize],
-    count_eq: &mut [usize],
-    edges: &[TI],
-    x: &[TI],
-    x_start_idx: usize,
-) {
-    if edges.is_empty() {
-        return;
-    }
-    if edges.len() == 1 {
-        let (num_lt, num_eq) = count_lt_eq(x, &edges[0]);
-        count_lt[0] = x_start_idx + num_lt;
-        count_eq[0] = num_eq;
-        return;
-    }
-    // use binary search to find #(x < middle edge) = |{i; x[i] < middle edge}|
-    let mid_edge_idx = (edges.len() + 1) / 2;
-    let mid_edge = &edges[mid_edge_idx];
-    let (num_lt, num_eq) = count_lt_eq(x, mid_edge);
-    count_lt[mid_edge_idx] = x_start_idx + num_lt;
-    count_eq[mid_edge_idx] = num_eq;
+pub(crate) fn score_candidates<TIA: PartialOrd>(
```

Comment:
This logic comes from the Polars implementation of this scorer. We had two different equivalent impls, now we have just one.